### ISSUE TITLE
CA2012 false positive for null conditional

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#3384]
#### Package Version

3.0.0-beta2.20167.4

#### Diagnostic ID

CA2012 (Use ValueTasks correctly)

#### Repro steps

https://github.com/dotnet/roslyn/blob/e0a1787d544c8e7c5fef2007dbc8ef6592d8c028/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/AsyncSymbolVisitor%601.cs#L15

```csharp
return symbol?.Accept(this) ?? new ValueTask<TResult>(DefaultResult);
```

#### Expected behavior

No diagnostic.

#### Actual behavior

> CA2012: ValueTask instances returned from method calls should be directly awaited, returned, or passed as an argument to another method call. Other usage, such as storing an instance into a local or a field, is likely an indication of a bug, as ValueTask instances must only ever be consumed once.

**Comment by stephentoub** [comment]
ISymbol.Accept returns a ValueTask?

**Comment by sharwell** [comment]
@stephentoub yes that is correct

**Comment by stephentoub** [comment]
Ok.  As an aside, "AcceptAsync" ;)
